manim/cli/checkhealth/checks.py

```diff
@@ -0,0 +1,173 @@
+
+    Returns
+    -------
+    A check function, as required by the checkhealth subcommand.
+    """
+    if skip_on_failed is None:
+        skip_on_failed = []
+    skip_on_failed = [
+        skip.__name__ if callable(skip) else skip for skip in skip_on_failed
+    ]
+
+    def decorator(func):
+        func.description = description
+        func.recommendation = recommendation
+        func.skip_on_failed = skip_on_failed
+        func.post_fail_fix_hook = post_fail_fix_hook
+        HEALTH_CHECKS.append(func)
+        return func
+
+    return decorator
+
+
+@healthcheck(
+    description="Checking whether manim is on your PATH",
+    recommendation=(
+        "The command <manim> is currently not on your system's PATH.\n\n"
+        "You can work around this by calling the manim module directly "
+        "via <python -m manim> instead of just <manim>.\n\n"
+        "To fix the PATH issue properly: "
+        "Usually, the Python package installer pip issues a warning "
+        "during the installation which contains more information. "
+        "Consider reinstalling manim via <pip uninstall manim> "
+        "followed by <pip install manim> to see the warning again, "
+        "then consult the internet on how to modify your system's "
+        "PATH variable."
+    ),
+)
+def is_manim_on_path():
+    path_to_manim = shutil.which("manim")
+    return path_to_manim is not None
+
+
+@healthcheck(
+    description="Checking whether the executable belongs to manim",
+    recommendation=(
+        "The command <manim> does not belong to your installed version "
+        "of this library, it likely belongs to manimgl / manimlib.\n\n"
+        "Run manim via <python -m manim> or via <manimce>, or uninstall "
+        "and reinstall manim via <pip install --upgrade "
+        "--force-reinstall manim> to fix this."
+    ),
+    skip_on_failed=[is_manim_on_path],
+)
+def is_manim_executable_associated_to_this_library():
+    path_to_manim = shutil.which("manim")
+    with open(path_to_manim, "rb") as f:
+        manim_exec = f.read()
+
+    # first condition below corresponds to the executable being
+    # some sort of python script. second condition happens when
+    # the executable is actually a Windows batch file.
+    return b"manim.__main__" in manim_exec or b'"%~dp0\\manim"' in manim_exec
+
+
+@healthcheck(
+    description="Checking whether ffmpeg is available",
+    recommendation=(
+        "Manim does not work without ffmpeg. Please follow our "
+        "installation instructions "
+        "at https://docs.manim.community/en/stable/installation.html "
+        "to download ffmpeg. Then, either ...\n\n"
+        "(a) ... make the ffmpeg executable available to your system's PATH,\n"
+        "(b) or, alternatively, use <manim cfg write --open> to create a "
+        "custom configuration and set the ffmpeg_executable variable to the "
+        "full absolute path to the ffmpeg executable."
+    ),
+)
+def is_ffmpeg_available():
```

Comment:
Unsure. I lean towards not splitting it; the case where `ffmpeg` is on the PATH but not executable is somewhat of corner case where someone on Linux or MacOS needs to explicitly mess with the executable flag on the script. I'm happy to split it if you feel strongly about it (but for the sake of consistency the other checks should then probably be split too).

Reply:
I just remember having some issues in the past with exactly that problem because ffmpeg was installed with a different user and therefore existed but was not accessible and executable. But we can leave and see how it goes in the future